erts/emulator/beam/sys.h

```diff
@@ -177,6 +177,23 @@ typedef ERTS_SYS_FD_TYPE ErtsSysFdType;
 #  define ERTS_UNLIKELY(BOOL) (BOOL)
 #endif
 
+#if (ERTS_AT_LEAST_GCC_VSN__(5, 1, 0) || __has_builtin(__builtin_unreachable))
+#  define ERTS_UNREACHABLE __builtin_unreachable()
+#elif defined(_MSC_VER)
+#  define ERTS_UNREACHABLE __assume(0)
+#else
+/* Unsupported compiler, just ignore it. */
+#  define ERTS_UNREACHABLE ((void)0)
+#endif
+
+#if !defined(DEBUG)
+#  define ERTS_ASSUME(Expr) ((Expr) ?                                          \
```

Comment:
Maybe a short comment about the purpose of ERTS_ASSUME. It's a compile time only thing on 'opt', right?